File: parallel.py
from concurrent.futures import ProcessPoolExecutor
from multiprocessing import get_context
from math import sqrt
from time import perf_counter

def f(n):
    s = 0.0
    for i in range(n):
        s += sqrt(i)
    return s

if __name__ == "__main__":
    inputs = [5_000_000] * 8

    start = perf_counter()
    a = list(map(f, inputs))
    t1 = perf_counter() - start

    start = perf_counter()
    with ProcessPoolExecutor(
        max_workers=8, mp_context=get_context("spawn")
    ) as pool:
        b = list(pool.map(f, inputs))
    tp = perf_counter() - start

    print(f"Sequential: {t1:.3f} s; parallel: {tp:.3f} s")
    print(f"Speedup: {t1 / tp:.2f}; same results: {a == b}")

File: parallel.mjs
import { Worker, isMainThread, parentPort, workerData }
  from 'node:worker_threads';

function f(n) {
  let s = 0;
  for (let i = 0; i < n; i++) s += Math.sqrt(i);
  return s;
}

function inWorker(n) {
  return new Promise((resolve, reject) => {
    const worker = new Worker(new URL(import.meta.url), {
      workerData: n
    });
    worker.once('message', resolve);
    worker.once('error', reject);
    worker.once('exit', code => {
      if (code !== 0) reject(new Error(`Worker exit: ${code}`));
    });
  });
}

if (!isMainThread) {
  parentPort.postMessage(f(workerData));
} else {
  const inputs = Array(8).fill(500_000_000);

  let start = performance.now();
  const a = inputs.map(f);
  const t1 = performance.now() - start;

  start = performance.now();
  const b = await Promise.all(inputs.map(inWorker));
  const tp = performance.now() - start;

  console.log(`Sequential: ${t1.toFixed(1)/1000} s`);
  console.log(`Parallel: ${tp.toFixed(1)/1000} s`);
  console.log('Speedup:', t1 / tp);
  console.log('Same results:', a.every((x, i) => x === b[i]));
}